pkg/cluster/cluster.go

```diff
@@ -37,33 +38,32 @@ const (
 	PeerTypeQuery = "query"
 )
 
-// NewPeer creates a new peer.
-func NewPeer(
-	l log.Logger,
-	name string,
-	typ PeerType,
-	addr string,
-	advertiseAddr string,
-	apiAddr string,
-	knownPeers []string,
-) (*Peer, error) {
+type JoinConfig struct {
+	Name          string
+	Addr          string
+	AdvertiseAddr string
+	KnownPeers    []string
+}
```

Comment:
I know this is very much a matter of personal taste, but I've recently come to prefer constructors with a few more arguments over creating many small config structs.  

To me it's mostly about readability, i.e. if I come back to this code in a few months I don't have to hop around for every constructor to figure out what arguments it is actually taking.
In this case here it's two of those config structs already. Then we have the indirection  in the main package as well, where we first create a config struct, pass it elsewhere, and then join the cluster. Before it was just a single step.

Summed up, `PeerState` sounds reasonable to me since it will probably grow over time. I also like `NewPeer` being renamed to `Join`.  The join arguments I'd rather keep inlined and have on step less in package `main`.

But as I said, it's mostly personal preference, so I just wanted to elaborate my view. But we can proceed as it is right now.

Reply:
This totally makes sense, I just wanted to avoid more arguments in function, but I guess you're right there is no harm in it.